sqlx uses chrono instead of time

### Bug Description
SQLx uses `chrono` types even if `time` feature is specified.

In `Cargo.toml`:

```toml
sqlx = { version = "0.7", features = [ "runtime-tokio",  "postgres", "tls-rustls", "uuid", "time" ] }
```

Column in Postgres:

```sql
created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
```

Rust model:

```rust
#[derive(sqlx::FromRow, Clone, Debug)]
struct Entry {
    ...
    pub created_at: time::OffsetDateTime,
}
```

Query:

```sql
SELECT
    ...    
    created_at
FROM entries
LIMIT 1
```

When I run `cargo sqlx prepare`, it finishes without errors.
Then, when I compile the binary, I get the following error:

```
error[E0277]: the trait bound `OffsetDateTime: From<DateTime<Utc>>` is not satisfied
the trait `From<DateTime<Utc>>` is not implemented for `OffsetDateTime`
```

As soon as I replace `time::OffsetDateTime` with `chrono::DateTime<Utc>` in the Rust type without changing anything else, the binary compiles.

### Info
* SQLx version: `0.7.1`
* SQLx features enabled: `"runtime-tokio",  "postgres", "tls-rustls", "uuid", "time"`
* Database server and version: `Postgres 15.2`
* Operating system: `macOS 13.4`
* `rustc --version`: `rustc 1.70.0 (90c541806 2023-05-31)`

--- a/sqlx-macros-core/src/database/postgres.rs
+++ b/sqlx-macros-core/src/database/postgres.rs
@@ -26,19 +26,19 @@ impl_database_ext! {
         #[cfg(feature = "uuid")]
         sqlx::types::Uuid,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::types::chrono::NaiveTime,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::types::chrono::NaiveDate,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::types::chrono::NaiveDateTime,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::types::chrono::DateTime<sqlx::types::chrono::Utc> | sqlx::types::chrono::DateTime<_>,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::postgres::types::PgTimeTz<sqlx::types::chrono::NaiveTime, sqlx::types::chrono::FixedOffset>,
 
         #[cfg(feature = "time")]
@@ -91,16 +91,16 @@ impl_database_ext! {
         #[cfg(feature = "uuid")]
         Vec<sqlx::types::Uuid> | &[sqlx::types::Uuid],
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         Vec<sqlx::types::chrono::NaiveTime> | &[sqlx::types::chrono::NaiveTime],
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         Vec<sqlx::types::chrono::NaiveDate> | &[sqlx::types::chrono::NaiveDate],
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         Vec<sqlx::types::chrono::NaiveDateTime> | &[sqlx::types::chrono::NaiveDateTime],
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         Vec<sqlx::types::chrono::DateTime<sqlx::types::chrono::Utc>> | &[sqlx::types::chrono::DateTime<_>],
 
         #[cfg(feature = "time")]
@@ -141,13 +141,13 @@ impl_database_ext! {
         #[cfg(feature = "rust_decimal")]
         sqlx::postgres::types::PgRange<sqlx::types::Decimal>,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::postgres::types::PgRange<sqlx::types::chrono::NaiveDate>,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::postgres::types::PgRange<sqlx::types::chrono::NaiveDateTime>,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::postgres::types::PgRange<sqlx::types::chrono::DateTime<sqlx::types::chrono::Utc>> |
             sqlx::postgres::types::PgRange<sqlx::types::chrono::DateTime<_>>,
 
@@ -173,19 +173,19 @@ impl_database_ext! {
         Vec<sqlx::postgres::types::PgRange<sqlx::types::Decimal>> |
             &[sqlx::postgres::types::PgRange<sqlx::types::Decimal>],
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         Vec<sqlx::postgres::types::PgRange<sqlx::types::chrono::NaiveDate>> |
             &[sqlx::postgres::types::PgRange<sqlx::types::chrono::NaiveDate>],
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         Vec<sqlx::postgres::types::PgRange<sqlx::types::chrono::NaiveDateTime>> |
             &[sqlx::postgres::types::PgRange<sqlx::types::chrono::NaiveDateTime>],
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         Vec<sqlx::postgres::types::PgRange<sqlx::types::chrono::DateTime<sqlx::types::chrono::Utc>>> |
             Vec<sqlx::postgres::types::PgRange<sqlx::types::chrono::DateTime<_>>>,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         &[sqlx::postgres::types::PgRange<sqlx::types::chrono::DateTime<sqlx::types::chrono::Utc>>] |
             &[sqlx::postgres::types::PgRange<sqlx::types::chrono::DateTime<_>>],
 

--- a/sqlx-macros-core/src/database/sqlite.rs
+++ b/sqlx-macros-core/src/database/sqlite.rs
@@ -12,13 +12,13 @@ impl_database_ext! {
         String,
         Vec<u8>,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::types::chrono::NaiveDate,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::types::chrono::NaiveDateTime,
 
-        #[cfg(feature = "chrono")]
+        #[cfg(all(feature = "chrono", not(feature = "time")))]
         sqlx::types::chrono::DateTime<sqlx::types::chrono::Utc> | sqlx::types::chrono::DateTime<_>,
 
         #[cfg(feature = "time")]

